tox.ini

```diff
@@ -9,7 +9,7 @@ allowlist_externals =
 passenv =
 	FORCE_COLOR
 commands =
-	python3 -m pytest -p no:cacheprovider --cov --cov-append --cov-report=term-missing jsonpickle tests {posargs}
```

Comment:
LGTM! I do wonder though, why were we not using a cache provider before?

Reply:
Thanks for the review ~ I'll add a PR link shortly.

The only reason I disabled the cache was to avoid `.pycache*` directories on disk ~ there really wasn't a strong reason otherwise. Not having a cache is actually kinda detrimental for others reasons (e.g. it prevents us from being able to use pytest's `--last-failed` and `--failed-first` options) so there's no downside to enabling it.

I suppose I should probably file a bug against pytest-ruff to make it gracefully handle the case where someone uses that flag, but I'll save that for later.